repo/maintenance/maintenance_schedule.go

```diff
@@ -72,34 +73,38 @@ func getAES256GCM(rep repo.DirectRepository) (cipher.AEAD, error) {
 }
 
 // TimeToAttemptNextMaintenance returns the time when we should attempt next maintenance.
-func TimeToAttemptNextMaintenance(ctx context.Context, rep repo.DirectRepository, max time.Time) (time.Time, error) {
+// if the maintenance is not owned by this user, returns time.Time{}.
+func TimeToAttemptNextMaintenance(ctx context.Context, rep repo.DirectRepository) (time.Time, error) {
 	mp, err := GetParams(ctx, rep)
 	if err != nil {
 		return time.Time{}, errors.Wrap(err, "unable to get maintenance parameters")
 	}
 
+	// if maintenance is not owned by this user, do not run maintenance here.
+	if !mp.isOwnedByByThisUser(rep) {
+		return time.Time{}, nil
```

Comment:
I prefer if we return a permission error here instead of time.Time{} as it fails silently

Reply:
I thought about this twice after reading the full change, I think you can still return a `time.Time{}` object that `IsZero()` for when maintenance is not scheduled. So, I think it is better to stay consistent with that behavior here. No need to return the error, just zero time when not an owner is ok.

Reply:
ack